src/util/numerics/mpz.cpp

```diff
@@ -12,12 +13,30 @@ Author: Leonardo de Moura
 #include <vector>
 
 namespace lean {
-mpz::mpz(uint64 v):
-    mpz(static_cast<unsigned>(v)) {
+
+mpz::mpz(uint64 v) : mpz(static_cast<unsigned>(v)) {
     mpz tmp(static_cast<unsigned>(v >> 32));
     mpz_mul_2exp(tmp.m_val, tmp.m_val, 32);
     mpz_add(m_val, m_val, tmp.m_val);
 }
+mpz::mpz(int64 v) : mpz(static_cast<unsigned>(v)) {
+    mpz tmp(static_cast<signed int>(v >> 32));
+    mpz_mul_2exp(tmp.m_val, tmp.m_val, 32);
+    mpz_add(m_val, m_val, tmp.m_val);
+}
+
+template<> long long int mpz::get() const {
+    mpz high_m, low_m;
+    mpz_fdiv_r_2exp(low_m.m_val, m_val, 32);
+    mpz_fdiv_q_2exp(high_m.m_val, m_val, 32);
+    return static_cast<long long int>(high_m.get<unsigned>()) << 32 | low_m.get<unsigned>();
```

Comment:
Should these have a `lean_assert` like the other ones?

Reply:
Note that `lean_assert` is only called in debug builds.

Reply:
Added